src/Encamina.Enmarcha.SemanticKernel/Internals/SemanticKernelCosineStringSimilarityComparer.cs

```diff
@@ -0,0 +1,55 @@
+﻿using System.Numerics.Tensors;
+
+using Encamina.Enmarcha.AI.Abstractions;
+
+using Microsoft.SemanticKernel;
+using Microsoft.SemanticKernel.Embeddings;
+
+namespace Encamina.Enmarcha.SemanticKernel.Internals;
+
+#pragma warning disable SKEXP0001 // Type is for evaluation purposes only and is subject to change or removal in future updates. Suppress this diagnostic to proceed.
+
+/// <summary>
+/// An implementation of the <see cref="IStringSimilarityComparer"/> interface that uses cosine similarity algorithm, using semantic kernel to generate the embeddings.
+/// </summary>
+internal sealed class SemanticKernelCosineStringSimilarityComparer : IStringSimilarityComparer
+{
+    private readonly Kernel kernel;
+    private readonly ITextEmbeddingGenerationService textEmbeddingGenerationService;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="SemanticKernelCosineStringSimilarityComparer"/> class.
+    /// </summary>
+    /// <param name="kernel">A valid instance of a <see cref="Kernel"/>.</param>
+    public SemanticKernelCosineStringSimilarityComparer(Kernel kernel)
```

Comment:
I think the name is a tongue twister. Let's simplify it. It could just be `CosineStringSimilarityComparer`, the Semantic Kernel is not required because it is already in a `Semantic Kernel` namespace.